tezos/api/src/ocaml_conv/from_ocaml.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use super::{
+    FfiPath, OCamlBlockHash, OCamlContextHash, OCamlHash, OCamlOperationHash, OCamlProtocolHash,
+};
+use crate::ffi::{
+    Applied, ApplyBlockResponse, Errored, ForkingTestchainData, JsonRpcResponse,
+    OperationProtocolDataJsonWithErrorListJson, PrevalidatorWrapper, ValidateOperationResponse,
+    ValidateOperationResult,
+};
+use crypto::hash::{BlockHash, ContextHash, Hash, OperationHash, ProtocolHash};
+use tezos_messages::p2p::encoding::operations_for_blocks::{Path, PathLeft, PathRight};
+use znfe::{FromOCaml, OCamlInt, IntoRust, OCaml, OCamlBytes, OCamlInt32, OCamlList};
+
+macro_rules! from_ocaml_hash {
+    ($ocaml_name:ident, $rust_name:ident) => {
+        unsafe impl FromOCaml<$ocaml_name> for $rust_name {
+            fn from_ocaml(v: OCaml<$ocaml_name>) -> Self {
+                unsafe { v.field::<OCamlBytes>(0).into_rust() }
+            }
+        }
+    }
+}
+
+from_ocaml_hash!(OCamlHash, Hash);
+from_ocaml_hash!(OCamlOperationHash, OperationHash);
+from_ocaml_hash!(OCamlBlockHash, BlockHash);
+from_ocaml_hash!(OCamlContextHash, ContextHash);
+from_ocaml_hash!(OCamlProtocolHash, ProtocolHash);
+
+// Helper for easier mapping from OCaml to Rust records
+macro_rules! unpack_ocaml_block {
+    ($var:ident => $cons:ident {
+        $($field:ident : $ocaml_typ:ty),+ $(,)?
+    }) => {
+        let record = $var;
+        unsafe {
+            let mut current = 0;
+
+            $(
+                current += 1;
+                let $field = record.field::<$ocaml_typ>(current - 1).into_rust();
+            )+
+
+            $cons {
+                $($field),+
+            }
+        }
+    };
+}
+
+unsafe impl FromOCaml<ForkingTestchainData> for ForkingTestchainData {
```

Comment:
@tizoc 
What do you think about to put:
```
unsafe impl FromOCaml<ForkingTestchainData> for ForkingTestchainData {
    fn from_ocaml(v: OCaml<ForkingTestchainData>) -> Self {
```
also inside the macro unpack_ocaml_block ?

Reply:
Sounds good, I like that.

Reply:
Done by adding a new macro, looks much better now: https://github.com/simplestaking/tezedge/pull/277/commits/6d1c4d2cfdb1adef968c69ba4a2cbfa91831e76e

Reply:
cool, yes, yes, much better, looks like more declarative style like encodings :)